### ISSUE TITLE
Live regions not announced correctly

### ISSUE THREAD

**Issue by miksa1987** [nvaccess/nvda#11410]
Hi, I'm in the process of making a React app as accessible as possible. However, I'm noticing some weird behavior with NVDA - I have a live region which I just cannot get NVDA to announce correctly. On Voiceover, everything works smoothly with this.

There's a live chat, which has generated HTML code like this:
```
<div>
  <ul aria-live="polite"> <-- There are multiple of these depending on number of elements
    <li aria-hidden="true/false"> <--- There are many of these, some hidden from screen reader depending on the component, these are messages, timestamps etc, some of them should be read
      <div>
        <span>
          <div>
            <p aria-hidden="true">Visible message text</p>
            <p aria-hidden="false">Hidden visibility, but contains additional information to screen reader and should be read always</p>
         </div>
      </span>
      </div>
    </li>
  </ul>
</div>
```
Here, I'm using multiple aria-live containers which are spawned when there are enough messages on the former one, this is due to Voiceover stopping the reading when there are too many elements, and this solution works with it. It should be also noted that there is a PoseGroup component wrapping the li elements, it doesn't show on the generated HTML. The li elements should only read the messages, no timestamps etc, that's why they have aria-hidden. No matter what I tried, NVDA doesn't read any of these. Voiceover works perfectly with this kind of code. I have tried to convert the ul and li's to divs with no luck.

Then there's a form input component, which has an error label as a live region, goes like this:
```
<label>
  <span>Some text</span>
  <input/>
  <div aria-live="polite">
    <span>This span is conditionally shown if there's an error.</span>
  </div>
</label>
```
This component however, works perfectly like this on both Voiceover and NVDA, and I just cannot get my head around why the other won't. Sure, it's more complex, but shouldn't it work on NVDA too since Voiceover does it without any difficulty? Is there something I'm missing in my code?

### Steps to reproduce:
- Create a React App which renders a code like the example. The `<li>` tags should be mapped as an array inside the `<ul>`. The `<ul>` should also be inside an array, because there will be more than one of them.
- Make it add likewise `<li>` tags inside the `<ul>` indefinitely on key press or button click - I'm just selecting the button with arrow keys and clicking it with space. You could randomly set aria-hidden to true or false for these.
- When there's enough of `<li>` tags, create a new `<ul>` and keep adding `<li>` inside it on key press or button click.

### Actual behavior:
NVDA does not announce any of the items inside the `<ul>` live regions.

### Expected behavior:
NVDA should read every added `<li>` with aria-hidden="false" inside the `<ul>` live regions when added. 
Voiceover, in comparison, handles this combination. It reads the first items when the live region is created, and announces every added item with aria-hidden set to false.

#### NVDA version:
2020.1
#### Windows version:
10.0.16299 build 16299

#### Name and version of other software in use when reproducing the issue:
Firefox 78.0.2

#### Does the issue still occur after restarting your computer?
Yes.

**Comment by feerrenrut** [comment]
I can't help with React app. But if you can reproduce the issue in plain HTML and javascript then we can investigate. 

From your sample and description I created the following to get you started: https://codepen.io/reefturner/full/RwrdNzL

I tested this with the latest alpha build of NVDA (alpha-20566,dfbda72e) and Firefox (78.0.2), i couldn't find any issues.

**Comment by miksa1987** [comment]
@feerrenrut I'll try to experiment some later with a React app from scratch if I could get deeper into this.. Your example works fine for me as well.

Do you happen to have any knowledge what Voiceover does differently with live regions than NVDA? I got a pretty big React app with all kinds of stuff in it, and it would have to be made as accessible as possible.

**Comment by Adriani90** [comment]
cc: @jcsteh maybe you have also some thoughts here.
Might it be possible that the app fires some other events which prevent or distract NVDA from reporting the live region?

**Comment by jcsteh** [comment]
As @feerrenrut noted, it's impossible to debug this without a live test case, and it'd be better if that test case were as distilled as possible (raw HTMl/JS).

That said, one thing to keep in mind is that NVDA won't read a live region the first time it is shown or added to the DOM. There's some controversy surrounding this, but the justification is that a live region is an area in which *updates* should be read, and initial creation isn't an update of the region. If we read the region on creation, we'd potentially end up reading a huge amount of text on page load.

So, for example, this won't read anything because the ul gets created with its content:

`data:text/html,<ul hidden id="live" aria-live="polite"><li>foo</li></ul><script>setTimeout(function() { live.hidden = false; }, 1000);</script>`

On the flip side, this *will* read something because the ul already existed and gets updated:

`data:text/html,<ul aria-live="polite"><li hidden id="liveitem">foo</li></ul><script>setTimeout(function() { liveitem.hidden = false; }, 1000);</script>`

**Comment by miksa1987** [comment]
@jcsteh @feerrenrut I managed to roughly replicate the behaviour, try this: https://aria-test-nvda.netlify.app/
And the source here: https://github.com/miksa1987/aria-test-nvda

It's very barebones, but structure is same as with the React app I'm battling with. Interesting here is that NVDA reads "luetelmapallo" on those li tags which have aria-hidden set to false, which translates to list bullet. It does not read the contents, which in turn Voiceover does. Might be the display: none on the `<p>` that contains the text for screen reader?

**Comment by jcsteh** [comment]
aria-hidden="false" does not work with visually hidden content in Firefox. Whether this interpretation of aria-hidden="false" is spec compliant is controversial; see https://github.com/w3c/core-aam/issues/58.

Just to make things interesting, it partially works in Chrome, but not fully. In the below example, the paragraph is exposed, but its text isn't:

`data:text/html,<p aria-hidden="false" style="display: none;">blah</p>`

Long story short: you should not use aria-hidden="false" if you want compatibility with all browsers and platforms. This is called out in the ARIA spec as well.
